### PR TITLE
frontend/fix:#908 fixed the customer invoice

### PR DESCRIPTION
**Pull request by ashkriti**
…omer invoice

## Type of Change

- [x] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## Description
Added nightcharges for checking if it is applied on a ride.
Changed the UI for the invoice pdf (removing the Namma Yatri logo)
Filtered "EXTRA_DISTANCE_FARE" coming in the farebreakuplist from backend

## Motivation and Context
https://github.com/nammayatri/nammayatri/issues/908

## How did you test it?
Tested Manually

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

## Screenshot
Screenshot 2023-05-19 at 8 35 52 PM

### CODE REVIEW

**Comment by ashkriti** [comment]
> What's the required of this issue?

So currently we are not getting from the backend if the night charges are applied in a particular completed ride so in the Invoice there is a text for the night charges to be shown so created a function to check the ride created at time if it comes under night charges.

File: Frontend/ui-customer/src/Helpers/Utils.js [hunk 1]
```diff
@@ -437,11 +437,21 @@ export const shuffle = function (array) {
...
+        if(timeStr == "currentTime"){
+          var currentTimeString = moment(new Date()).format("HH:mm:ss");
+        return startTime < endTime ? between(currentTimeString, startTime, endTime) : between(currentTimeString, startTime, "23:59:59") || between(currentTimeString, "00:00:01", endTime);
+        }
+        else{
+          const time = moment(timeStr, 'h:mm A');
+          const startTime = moment('10:00 PM', 'h:mm A');
+          const endTime = moment('5:00 AM', 'h:mm A').add(1, 'day');
+          return time.isBetween(startTime, endTime, null, '[]');;
+        }
```
Comment (ashishsingh101) [inline Frontend/ui-customer/src/Helpers/Utils.js:451]: @ashkriti can you optimize this func, here you can directly send timeStr as HH:mm:ss and no need for if-else conditions

File: Frontend/ui-customer/src/Screens/CustomerUtils/MyRidesScreen/Controller.purs [hunk 2]
```diff
@@ -200,10 +200,12 @@ myRideListTransformer state listRes = filter (\item -> (item.status == "COMPLETE
     fares = getFares ride.fareBreakup 
     (RideAPIEntity rideDetails) = (fromMaybe dummyRideAPIEntity (ride.rideList !!0))
     baseDistanceVal = (getKmMeter (fromMaybe 0.0 (rideDetails.chargeableRideDistance)))
+    timeVal = (convertUTCtoISC (fromMaybe ride.createdAt ride.rideStartTime) "h:mm A")
+    nightChargesVal = (withinTimeRange "10:00 PM" "5:00 AM" timeVal)
```
Comment (ashishsingh101) [inline Frontend/ui-customer/src/Screens/CustomerUtils/MyRidesScreen/Controller.purs:204]: you can send start time, end time and timeVal as HH:mm:ss (24 hrs) format.

**Review by ashishsingh101 — CHANGES_REQUESTED** [review]

File: Frontend/ui-customer/src/Helpers/Utils.js (hunk 1 above)
Comment (ashkriti) [reply Frontend/ui-customer/src/Helpers/Utils.js]: Done

File: Frontend/ui-customer/src/Screens/CustomerUtils/MyRidesScreen/Controller.purs (hunk 2 above)
Comment (ashkriti) [reply Frontend/ui-customer/src/Screens/CustomerUtils/MyRidesScreen/Controller.purs]: Done

**Review by ashishsingh101 — APPROVED** [review]

**Pull request merged**